File: src/main/webapp/js/services/disk_service.js
'use strict';

App.factory('DiskService', ['$http', '$q', function ($http, $q) {
    return {
        fetchAllUsers: function () {
            return $http.get('/getUsers')
                .then(
                function (response) {
                    return response.data;
                },
                function (errResponse) {
                    console.error('error');
                    return $q.reject(errResponse);
                }
            )
        },

        getFreeDisks: function () {
            return $http.get('/getFreeDisks')
                .then(
                function (response) {
                    return response.data;
                },
                function (errResponse) {
                    console.error('error');
                    return $q.reject(errResponse);
                }
            )
        },
        getTakenByMeDisks: function (id) {
            return $http({
                method: 'GET',
                url: '/getTakenDisks',
                params: {id: id}
            }).then(
                function (responce) {
                    return responce.data
                },
                function (errResponse) {
                    console.error('error');
                    return $q.reject(errResponse);
                }
            )
        },

        getGivenDisks: function (user_id) {
            return $http({
                method: 'GET',
                url: '/getGivenDisks',
                params: {user_id: user_id}
            }).then(
                function (response) {
                    return response.data
                },
                function (errResponse) {
                    console.error('error');
                    return $q.reject(errResponse);
                }
            )
        },

        rentDisk: function (disk_id, user_id) {
            return $http({
                method: 'POST',
                url: '/rentDisk',
                params: {disk_id: disk_id, user_id: user_id}
            });
        },
        returnDisk: function (disk_id, user_id) {
            return $http({
                method: 'POST',
                url: '/returnDisk',
                params: {disk_id: disk_id, user_id: user_id}
            });
        },
        addDisk: function (disk, owner_id) {
            return $http({
                method: 'POST',
                url: '/addDisk',
                params: {title: disk.title, rating: disk.rating, description: disk.description, owner_id: owner_id}
            });
        },
        getLogin: function (email, password) {
            return $http({
                method: 'GET',
                url: "/getLogin",
                params: {email: email, password: password}
            }).then(
                function (response) {
                    return response.data
                },
                function (errResponse) {
                    console.error('error');
                    return $q.reject(errResponse);
                }
            )
        }
    }
}]);

File: src/main/java/ru/datatex/service/DiskService.java
package ru.datatex.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.datatex.database.DiskDataProvider;
import ru.datatex.model.Disk;
import ru.datatex.model.Users;

import java.util.List;

@Service
public class DiskService {

    @Autowired
    DiskDataProvider diskDataProvider;

    public List<Users> getUsers() {
        return diskDataProvider.getAllUsers();
    }

    public List<Disk> getFreeDisks(Long id) {
        return diskDataProvider.getFreeDisks(id);
    }

    public void addDisk(String title, String rating, String description, Long ownerId) {
        diskDataProvider.addDisk(title, rating, description, ownerId);
    }

    public List<Disk> getTakenDisks(Long id) {
        return diskDataProvider.getTakenDisk(id);
    }

    public void rentDisk(Long diskId, Long userId) {
        diskDataProvider.rentDisk(diskId, userId);
    }

    public void returnDisk(Long diskId, Long userId) {
        diskDataProvider.returnDisk(diskId, userId);
    }

    public List<Disk> getGivenDisks(Long userId) {
        return diskDataProvider.getGivenDisks(userId);
    }

}
